components/datetime/src/format/time_zone.rs

```diff
@@ -54,258 +54,6 @@ where
     T: TimeZoneInput,
```

Comment:
Do we want to remove this whole file? 

Ideally, `format` or `format_with_fallback` should replace it instead.

Reply:
Hmm, well I think `format()` and `format_with_fallback()` would still produce a `FormattedTimeZone`.

The `FormattedTimeZone` would know whether or not it should call `format()` or `format_with_fallback()`.

I don't think we want to stray from the pattern of creating a `FormattedTimeZone` object, because that is congruent with how `DateTime` works.

I think perhaps the `new_formatted_time_zone()` constructor should take an optional fallback. And this will determine internally whether the `FormattedTimeZone` calls `format()` or `format_with_fallback()`.

Reply:
Because `FormattedTimeZone` has a `TimeZoneFormat`, we can use `TimeZoneFormat` in `DateTime` directly if `FormattedTimeZone` is just a simple wrapper.

I feel like we don't need `new_formatted_time_zone`. We can pass the fallback enum to the `try_new` or `try_from_config` in `TimeZoneFormat`.